docs/docs/aztec/concepts/accounts/index.md

```diff
@@ -72,87 +63,103 @@ def entryPoint(payload):
 
-Note that nothing related to signature verification or payload execution is enshrined in the protocol, since account contracts are free to define this entrypoint however they see fit. Therefore, a request for executing an action has a simpler structure than in Ethereum, and just requires:
+For example, a lottery contract, where at some point a prize needs to be paid out to its winners. This `pay` action does not require authentication and does not need to be executed by any user in particular, so anyone could submit a transaction that defines the lottery contract itself as `origin` and `pay` as `entrypoint` function.
 
-- The `origin` contract to execute as the first step
-- The initial function to call (usually `entrypoint`)
-- The arguments (which encode the private and public calls to run as well as any signatures)
+For an example of this behavior see our [non_contract_account test](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/end-to-end/src/e2e_non_contract_account.test.ts) and the [SignerLess wallet](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec.js/src/wallet/signerless_wallet.ts) implementation. Notice that the Signerless wallet doesn't invoke an `entrypoint` function of an account contract but instead invokes the target contract function directly.
 
-### Entrypoint restrictions
+:::info
+Entrypoints for the following cases:
 
-Entrypoint methods are not enshrined in the protocol, and any function can be called as an entrypoint. The only restriction is that it must be private (not open), so all transactions are initiated with a client-side zero-knowledge proof.
+- If no contract `entrypoint` is used `msg_sender` is set to `Field.max`.
+- In a private to public `entrypoint`, `msg_sender` is the contract making the private to public call.
+- When calling the `entrypoint` on an account contract, `msg_sender` is set to the account contract address.
+:::
 
-This means that, unlike other protocols, Aztec does not impose any restrictions on the actions that can be carried out during the validation phase of the entrypoint, since these actions are executed by the client and wrapped in a zero-knowledge proof that is verified by the network. You are free to call into other contracts, access storage, or do as much computing as you need in the entrypoint.
+### Account contracts and wallets
 
-### Nonces and replay protection
+Account contracts are tightly coupled to the wallet software that users use to interact with the protocol. Dapps submit to the wallet software one or more function calls to be executed (e.g. "call swap in X contract"), and the wallet encodes and authenticates the request as a valid payload for the user's account contract. The account contract then validates the request encoded and authenticated by the wallet, and executes the function calls requested by the dapp.
```

Comment:
Non-native speaker question: _authenticate_ here is used to mean "sign", but I've usually seen it used to mean "verify"...?